consensus/core/src/dag_state.rs

```diff
@@ -484,7 +539,12 @@ impl DagState {
                 .to_authority_index(authority_index)
                 .unwrap();
 
-            let last_evicted_round = self.authority_eviction_round(authority_index);
+            let last_evicted_round = if self.gc_enabled() {
+                self.authority_gc_eviction_round(authority_index)
```

Comment:
It seems cleaner to check for `self.gc_enabled()` inside `self.authority_eviction_round()` instead?